### DISCUSSION TITLE
Can I publish messages in batches to Rabbit MQ using Kombu?

### DISCUSSION THREAD

**Discussion by gcobr** [celery/kombu#1623] [Q&A] [answered]
I need to improve the performance of a batch process that needs to publish a very large number of messages to Rabbit MQ in the shortest period of time possible. I reckon publishing them one by one through a Producer might result in too many network roundtrips, and, thus, excessive networking overhead.

Is there a way to batch publish messages? For instance, would I be able to publish 10 or 100 messages at a time in a single network roundtrip?

**Comment by thedrow** [comment]
Not at the moment but we're currently working on such a feature to be released in 5.3

**Reply by alexgoryushkin** [reply]
Any news?

**Reply by Nusnus** [reply]
@alexgoryushkin 
> Any news?

It is off the table at the moment for 2023.
I will discuss it with @thedrow before the end of the year to prioritize it again.

**Reply by bajaj-hitesh** [reply]
Waiting for this feature. I have a similar requirement to publish large number of messages to RMQ, and batching would help

**Reply by alexgoryushkin** [reply]
Any news? 🥺

**Reply by Nusnus** [reply, marked as answer]
> Any news? 🥺

Unfortunately it was not prioritized at the end @alexgoryushkin 

We’ll welcome a contribution if someone is willing to make the effort though 🙏
